Packages/com.llealloo.audiolink/Editor/Scripts/AudioLinkPipelineDetector.cs

```diff
@@ -1,24 +1,72 @@
 using UnityEngine;
 using UnityEditor;
+using UnityEditor.Build;
+using UnityEditor.Build.Reporting;
 
 namespace AudioLink.Editor
 {
-    public static class AudioLinkPipelineDetector
+    public class AudioLinkPipelineDetector : IPreprocessBuildWithReport
     {
+        public int callbackOrder => 0;
+
+        private static readonly string[] s_shaderPrefixes = {
+            "AudioLink/StandardLit",
+            "AudioLink/Surface"
+        };
+
         [InitializeOnLoadMethod]
-        private static void DetectPackages()
+        private static void InitializeOnLoad()
+        {
+            EditorApplication.delayCall += SetShaderKeywords;
+        }
+
+        public void OnPreprocessBuild(BuildReport report)
         {
+            SetShaderKeywords();
+        }
+
+        private static void SetShaderKeywords()
+        {
+            string[] materialGuids = AssetDatabase.FindAssets("t:Material");
+            int count = 0;
+
+            foreach (string guid in materialGuids)
+            {
+                string path = AssetDatabase.GUIDToAssetPath(guid);
+                Material material = AssetDatabase.LoadAssetAtPath<Material>(path);
+
+                if (material != null && IsTargetShader(material.shader))
+                {
 #if URP_AVAILABLE
-            Shader.EnableKeyword("UNIVERSAL_RENDER_PIPELINE");
+                    material.EnableKeyword("_UNIVERSAL_RENDER_PIPELINE");
+                    material.DisableKeyword("_HIGH_DEFINITION_RENDER_PIPELINE");
+#elif HDRP_AVAILABLE
+                    material.EnableKeyword("_HIGH_DEFINITION_RENDER_PIPELINE");
```

Comment:
Is this intentional? You didn't add any HDRP support, right?

Reply:
No, HDRP hasn't been added yet. I was thinking of having it supported in the future, but we can remove that for now if you'd like.

Reply:
It can stay, nbd. Just wanted to make sure I wasn't missing something.